squire_lib/src/ffi_tournament.rs

```diff
@@ -86,6 +123,173 @@ impl TournamentId {
+                Some(mut v) => tournament = v.value(),
+                None => {
+                    println!("[FFI]: Cannot find tournament in update_settings");
+                    return false;
+                }
+            }
+        }
+
+        // Sort input strings out
+        let format =
+            String::from(unsafe { CStr::from_ptr(__format).to_str().unwrap().to_string() });
+
+        // Init list of operations to execute
+        let aid: AdminId = Uuid::default().into();
+        let mut op_vect: Vec<TournOp> = Vec::<TournOp>::new();
+        if format != tournament.format {
+            op_vect.push(TournOp::UpdateTournSetting(
+                aid,
+                TournamentSetting::Format(format),
+            ));
+        }
+
+        if starting_table_number != tournament.round_reg.starting_table {
+            op_vect.push(TournOp::UpdateTournSetting(
+                aid,
+                TournamentSetting::StartingTableNumber(starting_table_number),
+            ));
+        }
+
+        if use_table_number != tournament.use_table_number {
+            op_vect.push(TournOp::UpdateTournSetting(
+                aid,
+                TournamentSetting::UseTableNumbers(use_table_number),
+            ));
+        }
+
+        if game_size != tournament.game_size {
+            op_vect.push(TournOp::UpdateTournSetting(
+                aid,
+                TournamentSetting::PairingSetting::MatchSize(game_size),
+            ));
+        }
+
+        if min_deck_count != tournament.min_deck_count {
+            op_vect.push(TournOp::UpdateTournSetting(
+                aid,
+                TournamentSetting::MinDeckCount(min_deck_count),
+            ));
+        }
+
+        if max_deck_count != tournament.max_deck_count {
+            op_vect.push(TournOp::UpdateTournSetting(
+                aid,
+                TournamentSetting::MaxDeckCount(max_deck_count),
+            ));
+        }
+
+        if reg_open != tournament.reg_open {
+            op_vect.push(TournOp::UpdateReg(aid, reg_open));
+        }
+
+        if require_check_in != tournament.require_check_in {
+            op_vect.push(TournOp::UpdateTournSetting(
+                aid,
+                TournamentSetting::RequireCheckIn(require_check_in),
+            ));
+        }
+
+        if require_deck_reg != tournament.require_deck_reg {
+            op_vect.push(TournOp::UpdateTournSetting(
+                aid,
+                TournamentSetting::RequireDeckReg(require_deck_reg),
+            ));
+        }
+
+        // Apply all settings
+        let err: bool = false;
+        for i in 0..op_vect.len() {
+            match tournament.apply_op(op_vect[i]) {
+                Err(t_err) => {
+                    println!("[FFI]: update_settings error: {t_err}");
+                    err = true;
+                    break;
+                }
+                _ => {}
+            }
+        }
+
+        // Rollback on error.
+        if err {
```

Comment:
I don't think we need to rollback here. We just need to inform the user of which settings didn't get applied.